packages/core/src/service/services/order.service.ts

```diff
@@ -1532,17 +1532,21 @@ export class OrderService {
             .getMany();
 
         for (const inputLine of input.lines) {
-            const existingFulfillmentLine = existingFulfillmentLines.find(l =>
+            const fulfillmentLinesForOrderLine = existingFulfillmentLines.filter(l =>
                 idsAreEqual(l.orderLineId, inputLine.orderLineId),
             );
-            if (existingFulfillmentLine) {
+            if (fulfillmentLinesForOrderLine.length) {
+                const fulfilledQuantity = fulfillmentLinesForOrderLine.reduce(
```

Comment:
suggestion: We have a neat utility for this common operation, it's already imported in this file:

```ts
const fulfilledQuantity = summate(fulfillmentLinesForOrderLine, 'quantity');
```